src/Raven.Server/Documents/Includes/IncludeTimeSeriesCommand.cs

```diff
@@ -37,33 +46,138 @@ public void Fill(Document document)
                 if (Results.ContainsKey(docId))
                     continue;
 
-                var rangeResults = GetTimeSeriesForDocument(docId, kvp.Value);
-
-                Results.Add(docId, rangeResults);
-
+                var count = kvp.Value.Count(x => x.Name == Constants.TimeSeries.All);
```

Comment:
avoid Linq, by the way... kv.Value.Any is better because it looks for a first occurence

Reply:
it looks like you are doing this calculation for each document and it can be done once in ctor?

Reply:
refactored this code